src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CSharp.Analyzers/AllowHtmlAttributeAnalyzer.cs

```diff
@@ -62,11 +55,15 @@ private void AnalyzeAttribute(SyntaxNodeAnalysisContext context, INamedTypeSymbo
 
             // If the identifier resolves to an actual symbol that isn't System.Web.Mvc.AllowHtmlAttribute, then bail out
             var attrNameSymbol = context.SemanticModel.GetTypeInfo(attribute.Name).Type;
-            if (SymbolEqualityComparer.Default.Equals(attrNameSymbol, allowHtmlType))
+            if (attrNameSymbol is INamedTypeSymbol symbol
+                && attrNameSymbol is not IErrorTypeSymbol
+                && !symbol.ToDisplayString(NullableFlowState.NotNull).Equals(AllowHtmlAttributeName, StringComparison.Ordinal))
```

Comment:
@mjrousos Thanks for noting this. I reverted my changes in this area to the original code. However, I'm a bit confused how will the attribute syntax be resolved to its fully qualified name at this point?

Reply:
It's unlikely that we'll get an `INamedTypeSymbol` here. The scenarios in which that would happen are:

1. The `[AllowHtml]` attribute isn't actually referring to System.Web.Mvc.AllowHtmlAttribute but to some other unfortunately named attribute that we've mistaken for that one (maybe it's Foo.OtherAssembly.AllowHtmlAttribute). In that case, the symbol will likely resolve and we'll see that the attribute isn't the one we're looking for and not create a diagnostic.
2. Someone is using these analyzers and code fix providers on their own without running the upgrade assistant tool. That's perfectly fine and could result in us being able to resolve the AllowHtmlAttribute type symbol if the user is addressing that issue before they've addressed UA0001.

To try our best to cover all cases, we'll give a diagnostic for any use of [AllowHtml] where we can't resolve the syntax to a type symbol or where we can and see that it corresponds to System.Web.Mvc.AllowHtmlAttribute.